bootstrap/kubeadm/api/v1alpha4/kubeadmbootstrapconfig_types.go

```diff
@@ -333,5 +336,19 @@ type Filesystem struct {
 	ExtraOpts []string `json:"extraOpts,omitempty"`
 }
 
+// Proxy defines the http proxy settings used by container runtime during bootstrap
+type Proxy struct {
+	// HttpProxy specifies the proxy url for http traffic
+	HttpProxy string `json:"httpProxy,omitempty"`
+	// HttpsProxy specifies the proxy url for https traffic
+	HttpsProxy string `json:"httpsProxy,omitempty"`
+	// +optional
+	NoProxy []string `json:"noProxy,omitempty"`
+	// +optional
+	HttpProxyAuth *string `json:"httpProxyAuth,omitempty"`
+	// +optional
+	HttpsProxyAuth *string `json:"httpsProxyAuth,omitempty"`
```

Comment:
Is this meant to store credentials?
If yes, what about transforming this into a link to a secret, so credentials don't get spread into all the kubeadm config objects (assuming they are the same) and it is possible to define RBAC around their access/management?

Reply:
well, looking at the code below it seems this actually links to secrets, but it is not easy to understand this from the name/(missing) comments 😓